lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs

```diff
@@ -223,29 +230,67 @@ instance KnownNetwork n => DecodeAddress (Jormungandr n) where
                 <> B8.unpack (BS.pack [discriminant])
                 <> "."
 
--- | Generate a configuration file for Jörmungandr@0.3.999
+-- | Generate a configuration file for Jörmungandr@0.6
+-- Will throw "YamlException" or "IOException" when files cannot be
+-- read/written.
+genConfigFileYaml
+    :: FilePath
+    -- ^ State directory
+    -> PortNumber
+    -- ^ P2P port
+    -> BaseUrl
+    -- ^ Rest API base URL
+    -> Maybe FilePath
+    -- ^ User's config file
+    -> IO FilePath
+genConfigFileYaml stateDir addressPort restApiUrl baseConfig = do
+    let nodeConfigFile = stateDir </> "jormungandr-config.yaml"
+    base <- maybe (pure Aeson.Null) Yaml.decodeFileThrow baseConfig
+    genConfigFile stateDir addressPort restApiUrl base
+        & Yaml.encodeFile nodeConfigFile
+    pure nodeConfigFile
+
+-- | Create a Jormungandr config by first making some defaults, then adding the
+-- user's config file (if provided), then setting the API port to the value that
+-- we have chosen.
 genConfigFile
     :: FilePath
     -> PortNumber
     -> BaseUrl
     -> Aeson.Value
-genConfigFile stateDir addressPort (BaseUrl _ host port _) = object
-    [ "storage" .= (stateDir </> "chain")
-    , "rest" .= object
-        [ "listen" .= String listen ]
-    , "p2p" .= object
-        [ "trusted_peers" .= ([] :: [()])
-        , "topics_of_interest" .= object
-            [ "messages" .= String "low"
-            , "blocks" .= String "normal"
+    -> Aeson.Value
+genConfigFile stateDir addressPort (BaseUrl _ host port _) user =
+    mergeObjects [defaults, user, override]
+  where
+    defaults = object
+        [ "storage" .= (stateDir </> "chain")
+        , "p2p" .= object
+            [ "trusted_peers" .= ([] :: [()])
+            , "topics_of_interest" .= object
+                [ "messages" .= String "low"
+                , "blocks" .= String "normal"
+                ]
+            , "public_address" .= String publicAddress
             ]
-        , "public_address" .= String publicAddress
         ]
-    ]
-  where
+    override = object
+        [ "rest" .= object
+            [ "listen" .= String listen ]
+        ]
```

Comment:
Although silently overriding is convenient, I don't think we should be doing anything silently as it might become hard to debug for users. Especially because the config is never printed in the logs. 

What about returning a proper error if the `rest` field was present in the config something like:

"I can't use the given config file because I'd need to override some existing values. Please remove the 'rest' and let me generate one for you instead." 

Also, on a different note: do we need the `public_address` if we're running a passive node :thinking: ?